### PR TITLE
update platform docs for windows

### PR DESCRIPTION
**Pull request by arezaii**
This PR updates our platform specific documentation for Windows. Specifically it moves Cygwin to a section within Windows and adds a section above it for WSL. 

I have also added a small section to the GPU technote indicating it's possible to use NVidia GPUs through WSL.

TESTING:

- [x] generated docs and previewed Windows and GPU pages

[reviewed by @e-kayrakli - thanks!]

### CODE REVIEW

File: doc/rst/platforms/windows.rst [hunk 1]
```diff
@@ -1,8 +1,36 @@
...
+running Chapel on Windows, as it provides a more native Linux environment and
+better performance. The Cygwin approach is supported as a portability option
+and should not be expected to result in good performance.
+
+
+-------------------
+Using Chapel on WSL
+-------------------
+
+The Windows Subsystem for Linux (WSL) is a compatibility layer for running
```
Comment (e-kayrakli) [inline doc/rst/platforms/windows.rst:18]: Nit: Probably can just use WSL at this point.

File: doc/rst/platforms/windows.rst [hunk 2]
```diff
@@ -1,8 +1,36 @@
...
+Linux binary executables natively on Windows. It provides a Linux-compatible
+kernel interface developed by Microsoft, and it can run a variety of
```
Comment (e-kayrakli) [inline doc/rst/platforms/windows.rst:20]: Consider dropping "developed by Microsoft"

File: doc/rst/platforms/windows.rst [hunk 3]
```diff
@@ -1,8 +1,36 @@
...
+Linux distributions. For more information on WSL, see the
+`WSL documentation <https://docs.microsoft.com/en-us/windows/wsl/about>`_.
+
+To use Chapel on WSL, you will need to install a Linux distribution from the
+Microsoft Store. We recommend using Ubuntu, but others will likely work.
+Once you have installed your distribution, get the list of prerequisites
+from :ref:`readme-prereqs` and install Chapel as you would on a native Linux
+system. There are no platform specific settings for Chapel on WSL at this time.
```
Comment (e-kayrakli) [inline doc/rst/platforms/windows.rst:28]: hyphenate "platform-specific"

File: doc/rst/technotes/gpu.rst [hunk 4]
```diff
@@ -561,6 +561,20 @@ marked with * are covered in our nightly testing configuration.
   * Software:ROCm 4.2*, 4.4, 5.4
 
 
+GPU Support on WSL
```
Comment (e-kayrakli) [inline doc/rst/technotes/gpu.rst:564]: Spell out WSL

File: doc/rst/technotes/gpu.rst [hunk 5]
```diff
@@ -561,6 +561,20 @@ marked with * are covered in our nightly testing configuration.
...
+------------------------------------------------
+
+Starting with Chapel 2.0, we have demonstrated the ability to use NVIDIA GPUs
```
Comment (e-kayrakli) [inline doc/rst/technotes/gpu.rst:567]: Let's be more assertive. Consider:

> NVIDIA GPUs can be used through WSL.

as the first sentence.

File: doc/rst/technotes/gpu.rst [hunk 6]
```diff
@@ -561,6 +561,20 @@ marked with * are covered in our nightly testing configuration.
...
+through WSL. To enable GPU support on WSL we require the CUDA Toolkit to be
+installed in the WSL environment and the NVIDIA driver to be installed on the
+Windows host.
+See the `NVIDIA documentation
+<https://docs.nvidia.com/cuda/wsl-user-guide/index.html#getting-started-with-cuda-on-wsl-2>`_
+for more information on setting up CUDA on WSL.
+See `Using Chapel on WSL <../platforms/windows.html#using-chapel-on-wsl>`_
+for more information on using Chapel with WSL.
+
```
Comment (e-kayrakli) [inline doc/rst/technotes/gpu.rst:576]: ... but then maybe add:

> Note that this is not a configuration we test nightly. Please file a bug report if you encounter any issues.

You can link to: https://github.com/chapel-lang/chapel/issues/new

**Review by e-kayrakli — APPROVED** [review]

File: doc/rst/technotes/gpu.rst (hunk 4 above)
Comment (arezaii) [reply doc/rst/technotes/gpu.rst:564]: haha! it was originally and I changed it to this because it loooked long in the menu

**Comment by bradcray** [comment]
Thanks for picking this up, @arezaii!

**Pull request merged**